hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -308,7 +305,7 @@ def __init__(self, engine, initial, predicate, allow_transition):
         # testing and learning purposes.
         self.extra_dfas = {}
 
-    @derived_value
+    @derived_value  # type: ignore
```

Comment:
Would placing this on the definition work?

Reply:
It works for typing part, but fails in runtime:

```
  File "/Users/sobolev/Desktop/hypothesis/hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py", line 117, in <module>
    class Shrinker:
  File "/Users/sobolev/Desktop/hypothesis/hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py", line 310, in Shrinker
    def cached_calculations(self):
TypeError: 'staticmethod' object is not callable
```